core/conversion/conversion.cpp

```diff
@@ -291,8 +298,141 @@ void EvaluateLoopBlock(ConversionCtx* ctx, const torch::jit::Node* n) {
+
+    auto count_limit = loop->addTripLimit(*for_const->getOutput(0), nvinfer1::TripLimit::kCOUNT);
+    TRTORCH_CHECK(count_limit, "Unable to create trip limit layer from node: " << *n);
+    count_limit->setName("max trip limit layer");
+
+    // trip limit layer AND recurrent layer: loop condition
+    auto cond_weight = converters::Weights(ctx, (int32_t) (start_cond.toBool() ? 1 : 0));
+    auto while_const = ctx->net->addIdentity(*ctx->net->addConstant(cond_weight.shape, cond_weight.data)->getOutput(0));
+    TRTORCH_CHECK(while_const, "Unable to create identity layer from node: " << *n);
+    while_const->setOutputType(0, nvinfer1::DataType::kBOOL);    
+    
+    auto cond_limit = loop->addTripLimit(*while_const->getOutput(0), nvinfer1::TripLimit::kWHILE);
+    TRTORCH_CHECK(cond_limit, "Unable to create trip limit layer from node: " << *n);
+    cond_limit->setName("boolean condition trip limit layer");
+    //auto recurrent_cond = loop->addRecurrence(*while_const->getOutput(0));
+    //TRTORCH_CHECK(recurrent_cond, "Unable to create recurrent layer from node: " << *n);
+
+    //ctx->AssociateValueAndTensor(block->outputs()[0], recurrent_cond->getOutput(0));
+
+    // recurrent layer: trip_count 
+    auto trip_weight = converters::Weights(ctx, (int32_t) trip_count.toInt());
+    auto trip_const = ctx->net->addConstant(trip_weight.shape, trip_weight.data);
+    TRTORCH_CHECK(trip_const, "Unable to create constant layer from node: " << *n);
+    
+    auto recurrent_trip = loop->addRecurrence(*trip_const->getOutput(0));
+    TRTORCH_CHECK(recurrent_trip, "Unable to create recurrent layer from node: " << *n);
+    recurrent_trip->setName("recurrent layer for trip counter");
+
+    //ctx->AssociateValueAndTensor(block->inputs()[0], recurrent_trip->getOutput(0));
+
+    // add recurrent layers to loop
+    std::vector<nvinfer1::IRecurrenceLayer*> recurrent_tensors;
+
+    // loop through all recurrent inputs 
+    for (unsigned int i = 2; i < n->inputs().size(); i++) {
+        auto inp = n->inputs()[i];
+
+        if (inp->type()->isSubtypeOf(c10::TensorType::get())) {
+            auto recur = loop->addRecurrence(*ctx->value_tensor_map[inp]);
+            TRTORCH_CHECK(recur, "Unable to create recurrent layer from node: " << *n);
+
+            std::ostringstream tensor_id;
+            tensor_id << reinterpret_cast<int*>(ctx->value_tensor_map[inp]);
+
+            recur->setName(("recurrent layer for Tensor " + tensor_id.str()).c_str());
+            recurrent_tensors.push_back(recur);
+        } else {
+            TRTORCH_THROW_ERROR("Only recurrent Tensors allowed as input to Loop");
+        }
+    }
+
+    // evaluate/convert all nodes inside block
+    for (auto bn : block->nodes()) {
+        if (bn->kind() == torch::jit::prim::If) {
+            EvaluateConditionalBlock(ctx, bn, true);
+        } else if (evaluators::shouldEvalAtConversionTime(bn)) {
```

Comment:
Is there a reason we couldnt add support for nested loops both evaluated and converted  trivially here?

Reply:
"nested loops" - `prim::Loops` or `prim::If`? or both?